interceptor/translation_interceptor.go

```diff
@@ -0,0 +1,128 @@
+package interceptor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.temporal.io/server/common/api"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"google.golang.org/grpc"
+)
+
+type (
+	TranslationInterceptor struct {
+		logger      log.Logger
+		translators []Translator
```

Comment:
Basic idea is we could have a translator in this list for namespace name and another for cluster name translation